framework/OutStreams/PlotInterfaces/PopulationPlot.py

```diff
@@ -0,0 +1,155 @@
+    self.logVars    = spec.findFirst('logVars').value 
+    # checker; this should be superceded by "required" in input params
+    if self.sourceName is None:
+      self.raiseAnError(IOError, "Missing <source> node!")
+    if self.vars is None:
+      self.raiseAnError(IOError, "Missing <vars> node!")
+
+  def initialize(self, stepEntities):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, stepEntities, dict, contains all the Objects are going to be used in the
+                                current step. The sources are searched into this.
+      @ Out, None
+    """
+    src = self.findSource(self.sourceName, stepEntities)
+    if src is None:
+      self.raiseAnError(IOError, f'No source named "{self.sourceName}" was found in the Step for SamplePlot "{self.name}"!')
+    self.source = src
+    # sanity check
+    dataVars = self.source.getVars()
+    missing = [var for var in (self.vars) if var not in dataVars]
+    if missing:
+      msg = f'Source DataObject "{self.source.name}" is missing the following variables ' +\
+            f'expected by OptPath plotter "{self.name}": '
+      msg += ', '.join(f'"{m}"' for m in missing)
+      self.raiseAnError(IOError, msg)
+
+  def run(self):
+    """
+      Main run method.
+      @ In, None
+      @ Out, None
+    """   
+    data = self.source.asDataset().to_dataframe()
+    inVars  = self.source._inputs
+    outVars = self.source._outputs   
+    
+    nFigures = len(self.vars)
+    fig, axs = plt.subplots(nFigures,1)
+    fig.suptitle('Population Plot')
+
+    min_Gen = int(min(data['batchId']))
+    max_Gen = int(max(data['batchId']))
+    
+    for indexVar,var in enumerate(self.vars):
+        min_fit = np.zeros(max_Gen-min_Gen+1)
+        max_fit = np.zeros(max_Gen-min_Gen+1)
+        avg_fit = np.zeros(max_Gen-min_Gen+1)
+    
+        for idx,genID in enumerate(range(min_Gen,max_Gen+1,1)):
+            population = data[data['batchId']==genID]
+            min_fit[idx] = min(population[var])
+            max_fit[idx] = max(population[var])
+            avg_fit[idx] = population[var].mean()
+        
+        if var in inVars:
+          if var in self.logVars:
+            errorfill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='g', ax=axs[indexVar],logscale=True)
+          else:
+            errorfill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='g', ax=axs[indexVar])
+        else:
+          if var in self.logVars:
+            errorfill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='b', ax=axs[indexVar],logscale=True)
+          else:
+            errorfill(range(min_Gen,max_Gen+1,1), avg_fit, [min_fit,max_fit], color='b', ax=axs[indexVar])
+        axs[indexVar].set_ylabel(var)
+        if var == self.vars[-1]:
+          axs[indexVar].set_xlabel('Generation #')
+          
+    plt.savefig(f'{self.name}.png')
```

Comment:
same comments as provided in 'OptParallelCoordinatePlot'

Reply:
I agree, I have added possibility to choose from multiple formats using the <how> node